projects/fab-speed-dial/src/lib/fab-speed-dial.ts

```diff
@@ -182,18 +185,38 @@ export class EcoFabSpeedDialComponent implements AfterContentInit {
         }
     }
 
+    /**
+     * The autoclose feature when user clicks in document
+     */
+    @Input() get closeOnOutsideClick(): boolean {
+        return this._closeOnOutsideClick;
+    }
+
+    set closeOnOutsideClick(close: boolean) {
+        this._closeOnOutsideClick = coerceBooleanProperty(close);
+        if (!this.isInitialized) {
+            return;
+        }
+        this._processOutsideClickState();
+    }
+
     @Output() openChange: EventEmitter<boolean> = new EventEmitter<boolean>();
 
     @ContentChild(EcoFabSpeedDialActionsComponent) _childActions: EcoFabSpeedDialActionsComponent;
 
-    constructor(private elementRef: ElementRef, private renderer: Renderer2) {
+    constructor(private elementRef: ElementRef, private renderer: Renderer2, @Inject(DOCUMENT) private document: any) {
```

Comment:
Should probably be typed better:

```suggestion
    constructor(private elementRef: ElementRef, private renderer: Renderer2, @Inject(DOCUMENT) private document: Document) {
```

Reply:
When changing this i ran into [https://github.com/angular/angular/issues/20351](https://github.com/angular/angular/issues/20351)

Reply:
So i can set /** @dynamic */ comment before component and let 'Document' there or use 'any'.

Reply:
I guess `@dynamic` would do the trick indeed